ecs-agent/api/container/restart/restart_tracker_test.go

```diff
@@ -17,20 +17,34 @@
 package restart
 
 import (
+	"encoding/json"
+	"fmt"
 	"testing"
 	"time"
 
 	apicontainerstatus "github.com/aws/amazon-ecs-agent/ecs-agent/api/container/status"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
+// unmarshalStandardRestartPolicy allows us to verify that the restart policy unmarshalled
+// from JSON (as it comes in the task payload) is translated to expected behavior in
+// the restart tracker.
+func unmarshalStandardRestartPolicy(t *testing.T, ignoredCode int) RestartPolicy {
+	rp := RestartPolicy{}
+	jsonBlob := fmt.Sprintf(`{"enabled": true, "restartAttemptPeriod": 60, "ignoredExitCodes": [%d]}`, ignoredCode)
+	err := json.Unmarshal([]byte(jsonBlob), &rp)
+	require.NoError(t, err)
```

Comment:
> ... verify that the restart policy unmarshalled
> from JSON (as it comes in the task payload) is translated to expected behavior

nit (non-blocking): In addition to requiring no error, we could also require that `rp`'s enabled to be true, restart attempt period to be 60, and ignored exit codes to be `ignoredCode` as further verification of expected behavior.